src/Sarif/Writers/ResultLogJsonWriter.cs

```diff
@@ -245,10 +246,22 @@ public void WriteResults(IEnumerable<Result> results)
                 throw new ArgumentNullException(nameof(results));
             }
 
-            foreach (Result result in results)
+            if (results.Any())
             {
-                WriteResult(result);
+                foreach (Result result in results)
+                {
+                    WriteResult(result);
+                }
             }
+            else
+            {
+                if ((_writeConditions & Conditions.ResultsInitialized) != Conditions.ResultsInitialized)
+                {
+                    OpenResults();
+                }
+
+                CloseResults();
```

Comment:
>CloseResults [](start = 16, length = 12)

Must be under same condition as `OpenResults`.

Reply:
Come to think of it, the condition on L258 is always `true` in this branch, so you don't need the test.

---
In reply to: [432565758](https://github.com/microsoft/sarif-sdk/pull/1882#discussion_r432565758) [](ancestors = 432565758)